stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketClientConnectorListener.java

```diff
@@ -74,16 +81,38 @@ public void onMessage(WebSocketControlMessage webSocketControlMessage) {
 
     @Override
     public void onMessage(WebSocketCloseMessage webSocketCloseMessage) {
-        try {
-            WebSocketDispatcher.dispatchCloseMessage(connectionInfo, webSocketCloseMessage);
-        } catch (IllegalAccessException e) {
-            // Ignore as it is not possible have an Illegal access
+        if (connectionInfo.getWebSocketEndpoint().getMapValue(CLIENT_ENDPOINT_CONFIG).getMapValue(RETRY_CONFIG).
+                size() > 0) {
+            if (!reconnect(connectionInfo)) {
```

Comment:
`reconnect` function doesn't have a loop. How can we get a false here if `maxReconnectAttempts` is more than once?